include/wx/dc.h

```diff
@@ -1396,185 +1396,53 @@ class WXDLLIMPEXP_CORE wxDC : public wxObject
     wxDECLARE_NO_COPY_CLASS(wxDC);
 };
 
-// ----------------------------------------------------------------------------
-// helper class: you can use it to temporarily change the DC text colour and
-// restore it automatically when the object goes out of scope
-// ----------------------------------------------------------------------------
-
-class WXDLLIMPEXP_CORE wxDCTextColourChanger
-{
-public:
-    wxDCTextColourChanger(wxDC& dc) : m_dc(dc), m_colFgOld() { }
-
-    wxDCTextColourChanger(wxDC& dc, const wxColour& col) : m_dc(dc)
-    {
-        Set(col);
-    }
-
-    ~wxDCTextColourChanger()
-    {
-        if ( m_colFgOld.IsOk() )
-            m_dc.SetTextForeground(m_colFgOld);
-    }
-
-    void Set(const wxColour& col)
-    {
-        if ( !m_colFgOld.IsOk() )
-            m_colFgOld = m_dc.GetTextForeground();
-        m_dc.SetTextForeground(col);
-    }
-
-private:
-    wxDC& m_dc;
-
-    wxColour m_colFgOld;
-
-    wxDECLARE_NO_COPY_CLASS(wxDCTextColourChanger);
-};
-
-// ----------------------------------------------------------------------------
-// helper class: you can use it to temporarily change the DC text background colour and
-// restore it automatically when the object goes out of scope
-// ----------------------------------------------------------------------------
-
-class WXDLLIMPEXP_CORE wxDCTextBgColourChanger
-{
-public:
-    wxDCTextBgColourChanger(wxDC& dc) : m_dc(dc) { }
-
-    wxDCTextBgColourChanger(wxDC& dc, const wxColour& col) : m_dc(dc)
-    {
-        Set(col);
-    }
-
-    ~wxDCTextBgColourChanger()
-    {
-        if ( m_colBgOld.IsOk() )
-            m_dc.SetTextBackground(m_colBgOld);
-    }
-
-    void Set(const wxColour& col)
-    {
-        if ( !m_colBgOld.IsOk() )
-            m_colBgOld = m_dc.GetTextBackground();
-        m_dc.SetTextBackground(col);
-    }
-
-private:
-    wxDC& m_dc;
-
-    wxColour m_colBgOld;
-
-    wxDECLARE_NO_COPY_CLASS(wxDCTextBgColourChanger);
-};
+// wxDCXXXChanger implementation
+
+#define wxDC_CHANGER(classname, attr_type, const_attr_type,  \
```

Comment:
The macro argument `const_attr_type` could be removed if using `const int&` instead of `int` in ctor and `Set()` of `wxDCTextBgModeChanger` is OK (I see no reason why not).